kable-btleplug-ffi/src/characteristic.rs

```diff
@@ -0,0 +1,50 @@
+use crate::descriptor::Descriptor;
+use crate::uuid::Uuid;
+use btleplug::api::CharPropFlags;
+
+#[derive(Clone, uniffi::Record)]
+pub struct Characteristic {
+    pub uuid: Uuid,
+    pub service: Uuid,
+    pub properties: CharacteristicPropertyFlags,
+    pub descriptors: Vec<Descriptor>,
+}
+
+#[derive(Clone, uniffi::Record)]
+pub struct CharacteristicPropertyFlags {
+    bits: u8,
+}
+
+impl From<Characteristic> for btleplug::api::Characteristic {
+    fn from(value: Characteristic) -> Self {
+        btleplug::api::Characteristic {
+            uuid: value.uuid.into(),
+            service_uuid: value.service.into(),
+            properties: value.properties.into(),
+            descriptors: value.descriptors.into_iter().map(|x| x.into()).collect(),
```

Comment:
nit: no need for the lamda, should be able to pass the method directly:
```suggestion
            descriptors: value.descriptors.into_iter().map(Descriptor::into).collect(),
```

Reply:
Turns out there's a default-off lint enforcing this. I've enabled it for the project so there'll be no decision making in the future